hugo/content/playground/user-validator.ts

```diff
@@ -0,0 +1,28 @@
+import { AstNode, AstNodeDescription, AstUtils, DefaultDocumentValidator, DefaultLinker, DiagnosticInfo, DocumentState, DocumentValidator, LangiumDocument, LinkingError, LinkingErrorData, ReferenceInfo, ValidationOptions } from "langium";
+import { LangiumServices } from "langium/lsp";
+import { Diagnostic } from "vscode-languageserver";
+
+export class PlaygroundValidator extends DefaultDocumentValidator {
+    constructor(services: LangiumServices) {
+        super(services);
+    }
+    protected override processLinkingErrors(document: LangiumDocument, diagnostics: Diagnostic[], _options: ValidationOptions): void {
+        for (const reference of document.references) {
+            const linkingError = reference.error;
+            if (linkingError) {
+                const info: DiagnosticInfo<AstNode, string> = {
+                    node: linkingError.container,
+                    property: linkingError.property,
+                    index: linkingError.index,
+                    data: {
+                        code: DocumentValidator.LinkingError,
+                        containerType: linkingError.container.$type,
+                        property: linkingError.property,
+                        refText: linkingError.reference.$refText
+                    } satisfies LinkingErrorData
+                };
+                diagnostics.push(this.toDiagnostic('warning', `Linking failed. Please inject a custom scope provider. This is a limitation of the playground. To overcome this issue, please study the learning section in the Langium documentation (original error: ${linkingError.message})`, info));
```

Comment:
Suggestion: I believe it makes more sense to include the original error at the start of the diagnostic. Also asking the user to inject a custom scope provider is a bit weird in the context of the playground, as this isn't possible there.
```suggestion
                diagnostics.push(this.toDiagnostic('warning', ${linkingError.message}. In case you want to adjust the linking rules, please consult the learning section in the Langium documentation.`, info));
```

Reply:
Done, thank you.

Reply:
<img width="747" alt="Bildschirmfoto 2024-05-23 um 13 04 25" src="https://github.com/eclipse-langium/langium-website/assets/1525964/ddda1c96-d87a-4be0-9d17-ce3f970836d0">